pylxd/models/container.py

```diff
@@ -90,6 +91,27 @@ def get(self, filepath):
                 params={'path': filepath})
             return response.content
 
+        def recursive_put(self, src, dst):
+            if os.path.isdir(src):
+                idx = len(os.path.dirname(src))
+                dst_dirs = []
+                for path, dirname, files in os.walk(src):
+                    dst_dir = os.path.join(dst, path[idx:].lstrip(os.path.sep))
+                    dst_dirs.append(dst_dir)
+
+                self._container.execute(['mkdir', '-p'] + dst_dirs)
```

Comment:
`dst_dirs` might end up being very long if there are many directories and also lots of duplicate parents if it is deeply nested.  What's really needed here is just to know the "tips" that need to be created, rather than all of the intermediates.  Using a `collections.defaultdict(dict)` and then just using values that are not dicts would cut down on the line length.

Alternatively, make the directories as you go by just traversing once, but using the API to push a directory rather than a file.  (mod to file put needed)